open-apparel-registry/open-apparel-registry
# improved UX on facility list tables

## Overview

- Better highlighting/visualization of certain status'
- Expanded row content will now render in a new table row instead of as extra divs within columns. This will fix alignment issues and overflow issues.
- Fixed table layouts. Several columns now have `colspans` defined which prevent shrinking/expanding based on content.
- Removed truncation and tooltips from longer addresses and names. These are longer required.

Connects #XXX

## Demo

**Highlight areas which require specific actions:**
<img width="1804" alt="Screen Shot 2020-10-23 at 4 07 06 PM" src="https://user-images.githubusercontent.com/1928955/97193275-360c6700-177f-11eb-8197-b8bb212f0e66.png">

**Highlight rows which have expandable rows**
<img width="1804" alt="Screen Shot 2020-10-23 at 4 07 19 PM" src="https://user-images.githubusercontent.com/1928955/97193290-399fee00-177f-11eb-80c9-0b547de69c9b.png">
<img width="1804" alt="Screen Shot 2020-10-23 at 4 07 23 PM" src="https://user-images.githubusercontent.com/1928955/97193301-3d337500-177f-11eb-8ef0-334f3f49f8a2.png">

**Improved highlighting of rows with errors**
<img width="1804" alt="Screen Shot 2020-10-23 at 4 07 30 PM" src="https://user-images.githubusercontent.com/1928955/97193312-3efd3880-177f-11eb-8969-499ac6854f7d.png">
<img width="1804" alt="Screen Shot 2020-10-23 at 4 07 35 PM" src="https://user-images.githubusercontent.com/1928955/97193322-415f9280-177f-11eb-8e7b-8d1296cebb25.png">

**Deleted rows will now be crossed-out**
<img width="1804" alt="Screen Shot 2020-10-23 at 4 08 07 PM" src="https://user-images.githubusercontent.com/1928955/97193326-43295600-177f-11eb-9aed-4428732c8140.png">

**Deleted rows that are expandable will still be highlighted**
<img width="1804" alt="Screen Shot 2020-10-23 at 4 08 11 PM" src="https://user-images.githubusercontent.com/1928955/97193328-43c1ec80-177f-11eb-845b-08af1646a137.png">
<img width="1804" alt="Screen Shot 2020-10-23 at 4 08 15 PM" src="https://user-images.githubusercontent.com/1928955/97193332-43c1ec80-177f-11eb-86f1-4b92b61519c1.png">

**Long names/addresses will no longer truncate and tooltip.**
<img width="1804" alt="Screen Shot 2020-10-23 at 4 09 47 PM" src="https://user-images.githubusercontent.com/1928955/97193336-445a8300-177f-11eb-8c35-f653248da909.png">


## Notes

I was originally going to try and refactor the code-base for these layouts. This proved to be a time-consuming and confusing process. So instead of spending a ton of time trying to rewrite, I instead focused on usability. 

There is a ton of duplicate code and functionality, but I felt improving the UX was a better way of spending time.

One UX issues to note: Some rows can be "expanded" but they don't have any extra content to reveal. I never figured out how to resolve this. 

## Testing Instructions

* Open up facility list tables
* Things to check:
    * Long names and address' (should not truncate)
    * Deleted rows (should be crossed-out)
    * Errors (should be highlighted in red)
    * Potential Match (should be highlighted blue)
    * Expandable rows (should render new row when open. Alignment issues should be resolved. Titles and content will be lined up correctly)

## Checklist
